src/main/java/com/cloudant/client/api/CloudantClient.java

```diff
@@ -141,85 +132,17 @@
-
-        doInit(h.get("scheme"),
-                h.get("hostname"),
-                new Integer(h.get("port")).intValue(),
-                loginUsername,
-                password,
-                null //connectoptions
-        );
-    }
-
-    /**
-     * Constructs a new instance of this class and connects to the cloudant account with the
-     * specified credentials
-     *
-     * @param account        For cloudant.com, the cloudant account to connect to. For Cloudant
-     *                       local, the server URL
-     * @param loginUsername  The apiKey (if using an APIKey, else pass in the account for this
-     *                       parameter also)
-     * @param password       The Password credential
-     * @param connectOptions optional properties to connect e.g connectionTime,socketTimeout,etc
-     */
-    public CloudantClient(String account, String loginUsername, String password, ConnectOptions
-            connectOptions) {
-        super();
-        Map<String, String> h = parseAccount(account);
-
-        doInit(h.get("scheme"),
-                h.get("hostname"),
-                new Integer(h.get("port")).intValue(),
-                loginUsername,
-                password,
-                connectOptions
-        );
-    }
-
-    /**
-     * Constructs a new instance of this class and connects to the cloudant account with the
-     * specified credentials
-     *
-     * @param account For cloudant.com, the cloudant account to connect to. For Cloudant
-     *                local, the server URL
-     */
-    public CloudantClient(String account) {
-        super();
-        Map<String, String> h = parseAccount(account);
-
-        doInit(h.get("scheme"),
-                h.get("hostname"),
-                new Integer(h.get("port")).intValue(),
-                null, null, null
-        );
-    }
+    CloudantClient(CouchDbProperties props, GsonBuilder gsonBuilder) {
+        this.couchDbClient = new CouchDbClient(props);
 
-    /**
-     * Constructs a new instance of this class and connects to the cloudant account with the
-     * specified credentials
-     *
-     * @param account        For cloudant.com, the cloudant account to connect to. For Cloudant
-     *                       local, the server URL
-     * @param account        The cloudant account to connect to
-     * @param connectOptions optional properties to connect e.g connectionTime,socketTimeout,etc
-     */
-    public CloudantClient(String account, ConnectOptions connectOptions) {
-        super();
-        Map<String, String> h = parseAccount(account);
-
-        doInit(h.get("scheme"),
-                h.get("hostname"),
-                new Integer(h.get("port")).intValue(),
-                null, null, connectOptions);
+        // set a custom gsonbuilder that includes additional cloudant deserializers
+        if(gsonBuilder != null) {
+            couchDbClient.setGsonBuilder(gsonBuilder);

```

Comment:
The `setGsonBuilder` method in this class needs to be made private (or the content could potentially be moved into the `ClientBuilder.setGsonBuilder()` method instead).